File: File-Handling-Practice-Questions/Java-Programming/DisplayFileContents.java
/*
    Write Java program to accept file name from user and open that file
    and display the contents on screen.
*/

import java.io.*;
import java.util.Scanner;

class DisplayFileContents
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        String FileName = null;
        boolean bRet = false;

        System.out.println("Enter the file name : ");
        FileName = sobj.nextLine();

        bRet = FileX.DisplayFile(FileName);

        if(bRet == false)
        {
            System.out.println("Unable to open file");
        }
    }
}

class FileX
{
    public static boolean DisplayFile(String FileName)
    {
        byte Buffer[] = new byte[1024];
        int iRet = 0;

        try
        {
            FileInputStream fiobj = new FileInputStream(FileName);

            while((iRet = fiobj.read(Buffer)) != -1)
            {
                System.out.print(new String(Buffer, 0, iRet));
            }

            fiobj.close();

            return true;
        }
        catch(IOException obj)
        {
            return false;
        }
    }
}

File: File-Handling-Practice-Questions/C-Programming/display_file_contents.c
/*
    Write application which accept file name from user and read all data
    from that file and display contents on screen.

    Input : Demo.txt
    Output : Display all the data of file.
*/

#include<stdio.h>
#include<unistd.h>
#include<fcntl.h>

#define BUFFER_SIZE 1024
#define ERR_OPEN -1

int DisplayFile(char *FileName)
{
    char Buffer[BUFFER_SIZE];
    int fd = 0, iRet = 0;

    fd = open(FileName, O_RDONLY);

    if(fd == -1)
    {
        return ERR_OPEN;
    }

    while((iRet = read(fd, Buffer, sizeof(Buffer))) != 0)
    {
        write(1, Buffer, iRet);
    }

    close(fd);

    return 0;
}

int main()
{
    char Fname[30];
    int iRet = 0;

    printf("Enter the file name : ");
    scanf("%s", Fname);

    iRet = DisplayFile(Fname);

    if(iRet == ERR_OPEN)
    {
        printf("Unable to open file\n");
    }

    return 0;
}